includes/class-wc-validation.php

```diff
@@ -79,6 +79,8 @@ public static function is_postcode( $postcode, $country ) {
 			case 'IE':
 				$valid = (bool) preg_match( '/([AC-FHKNPRTV-Y]\d{2}|D6W)[0-9AC-FHKNPRTV-Y]{4}/', wc_normalize_postcode( $postcode ) );
 				break;
+			case 'IN':
+				$valid = (bool) preg_match( '/^([1-9]{6})$/', $postcode );		
```

Comment:
Hello,

For India we only the first digit will start with `1`, so we need to start eh regex with `[1-9]{1}`, after that it's allowed any number from `0-9`.

It will work better with the follow regex:

```suggestion
				$valid = (bool) preg_match( '/^[1-9]{1}[0-9]{2}\s{0,1}[0-9]{3}$/', $postcode );		
```

It also allows formatting, in case someone uses a space to format.

Reply:
Thanks a lot for this review and commit suggestion. This is a sincere request to REMOVE space to format. Space is a bane is most systems which try to build any form of automation. We should encourage standardization as far as possible.

Thanks  a lot.

Reply:
> Please revisit this understanding - only the first digit will start with 1. Actually FIRST DIGIT should NOT be zero. It can be any OTHER digit. If you force the first digit to be 1 it will be incorrect.

We aren't forcing to be `1`, the regex allows to be any number from `1` to `9`.

See all the regex:

- `[1-9]{1}` first digit will be any number from `1` to `9`
- `[0-9]{2}` Now allows 2 digits with any number from `0` to `9`
- `\s{0,1}` Allows 1 space, also accepts if none is entered
- `[0-9]{3}` And finally the last 3 digits from `1` to `9`.

For example both formats will be accepted: `282002` and `282 002`.

> This is a sincere request to REMOVE space to format. Space is a bane is most systems which try to build any form of automation. We should encourage standardization as far as possible.

This is for validation, while some system doesn't use space, it still is an allowed format, we shouldn't throw warnings if someone enters a space.

Reply:
See some tests that I made on WP CLI:

Incorrect regex:
```
wp> (bool) preg_match( '/^([1-9]{6})$/', '282002' )
=> bool(false)
```

Correct regex:

```
wp> (bool) preg_match( '/^[1-9]{1}[0-9]{2}\s{0,1}[0-9]{3}$/', '282002' )
=> bool(true)
wp> (bool) preg_match( '/^[1-9]{1}[0-9]{2}\s{0,1}[0-9]{3}$/', '282 002' )
=> bool(true)
wp> (bool) preg_match( '/^[1-9]{1}[0-9]{2}\s{0,1}[0-9]{3}$/', '082 002' )
=> bool(false)
```

Reply:
@claudiosanches I would sincerely request you to remove the permission to add space in the pin code

Please see sample records from India Post website https://www.indiapost.gov.in/vas/pages/LocatePostOffices.aspx?Pin=T6VBAueiFKw= 

as well as some sample records in the screenshot

![image](https://user-images.githubusercontent.com/454883/92201214-9a0a6180-ee99-11ea-8c67-fc90755196fc.png)

We should use system validation to prompt customers to use the correct format.

Reply:
@KoolPal there's nothing wrong accepting the space, WooCommerce there's functions to sanitize postcodes and remove all spaces if you are sending to a payment gateway or something like that.
Not allowing spaces will be a problem for users already used to enter postcodes with an space, showing to they a message saying that they postcode is incorrect. This is a huge issue for UX, and we can't allow. I hope you understand.

Reply:
@claudiosanches Sure. I understand. Let it be. Some validation is BETTER than no logic for Pin code for India